frontend/src/pages/modelServing/screens/metrics/performance/ModelGraphs.tsx

```diff
@@ -5,34 +5,117 @@ import {
   ModelMetricType,
   ModelServingMetricsContext,
 } from '~/pages/modelServing/screens/metrics/ModelServingMetricsContext';
-import { ContextResourceData, PrometheusQueryRangeResultValue } from '~/types';
+import {
+  ContextResourceData,
+  PrometheusQueryRangeResponseDataResult,
+  PrometheusQueryRangeResultValue,
+} from '~/types';
+import { InferenceServiceKind } from '~/k8sTypes';
+import { isModelMesh } from '~/pages/modelServing/utils';
+import { NamedMetricChartLine } from '~/pages/modelServing/screens/metrics/types';
+import {
+  convertPrometheusNaNToZero,
+  toPercentage,
+} from '~/pages/modelServing/screens/metrics/utils';
+
+type ModelGraphProps = {
+  model: InferenceServiceKind;
+};
 
-const ModelGraphs: React.FC = () => {
+const ModelGraphs: React.FC<ModelGraphProps> = ({ model }) => {
   const { data } = React.useContext(ModelServingMetricsContext);
+  const modelMesh = isModelMesh(model);
 
   return (
     <Stack hasGutter>
-      <StackItem>
-        <MetricsChart
-          metrics={[
-            {
-              name: 'Successful',
-              metric: data[
-                ModelMetricType.REQUEST_COUNT_SUCCESS
-              ] as ContextResourceData<PrometheusQueryRangeResultValue>,
-            },
-            {
-              name: 'Failed',
-              metric: data[
-                ModelMetricType.REQUEST_COUNT_FAILED
-              ] as ContextResourceData<PrometheusQueryRangeResultValue>,
-            },
-          ]}
-          color="blue"
-          title="HTTP requests per 5 minutes"
-          isStack
-        />
-      </StackItem>
+      {modelMesh ? (
```

Comment:
Here I think it would be good to just conditionally render a ModelMeshMetrics / KserveMetrics component so the logic can stay separate. In the kserve component I'd just put some kind of placeholder where the actual graph will be and I'll just add that in with my upcoming PR.